src/solvers/dg_p4est/containers_2d.jl

```diff
@@ -20,12 +20,20 @@ end
 # Interpolate tree_node_coordinates to each quadrant
 function calc_node_coordinates!(node_coordinates,
                                 mesh::P4estMesh{2},
-                                nodes)
+                                basis::LobattoLegendreBasis)
   # Hanging nodes will cause holes in the mesh if its polydeg is higher
   # than the polydeg of the solver.
-  @assert length(nodes) >= length(mesh.nodes) "The solver can't have a lower polydeg than the mesh"
-
-  tmp1 = zeros(real(mesh), 2, length(nodes), length(mesh.nodes))
+  @assert length(basis.nodes) >= length(mesh.nodes) "The solver can't have a lower polydeg than the mesh"
+
+  # We use `StrideArray`s here since these buffers are used in performance-critical
+  # places and the additional information passed to the compiler makes them faster
+  # than native `Array`s.
+  tmp1    = StrideArray(undef, real(mesh),
+                        StaticInt(2), static_length(basis.nodes), static_length(mesh.nodes))
+  matrix1 = StrideArray(undef, real(mesh),
+                        static_length(basis.nodes), static_length(mesh.nodes))
+  matrix2 = similar(matrix1)
```

Comment:
So similar has a method for `StrideArray` and is not just creating a plain `Array`, correct?

Reply:
Yes